### PR TITLE
Continued select case var and cleanup

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
 - Continued effort, working through remaining files.

### CODE REVIEW

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 1]
```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
                 // get inlet and outlet node number from equipment list
                 CurrentModuleObject = "AirLoopHVAC:OutdoorAirSystem:EquipmentList";
                 for (int CompNum = 1; CompNum <= state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).NumComponents; ++CompNum) {
-                    std::string CompType = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentType(CompNum);
-                    std::string CompName = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum);
+                    auto &CompType = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentType(CompNum);
```
Comment (Myoldmopar) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:471]: To get the full effect of this change, my suggestion is to look at both the develop [version](https://github.com/NREL/EnergyPlus/blob/c8eacd64f9f2ba2d7bd64c2617f4d2474436aefc/src/EnergyPlus/AirLoopHVACDOAS.cc#L399) and this branch [version](https://github.com/NREL/EnergyPlus/blob/22cc14d218f4fa28d074125c72783c27108d337a/src/EnergyPlus/AirLoopHVACDOAS.cc#L470) of this file side by side.  This is _a little_ better to read and understand now.  And DRYed up some duplicate blocks.

Comment (amirroth) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:471]: Don't reference integers or enums. You're giving the compiler chaff.

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 2]
```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
...
+                    auto &CompName = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum);
```
Comment (amirroth) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:472]: `string_view` and `int`, don't `auto` these. And don't reference `int`.

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 3]
```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
...
+                    auto &thisInletNodeNum = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).InletNodeNum(CompNum);
+                    auto &thisOutletNodeNum = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).OutletNodeNum(CompNum);
+
                     bool InletNodeErrFlag = false;
```
Comment (amirroth) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:476]: what, no `auto` here?

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 4]
```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
...
                     bool OutletNodeErrFlag = false;
 
-                    auto const SELECT_CASE_var(
-                        UtilityRoutines::MakeUPPERCase(state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentType(CompNum)));
+                    auto typeNameUC = UtilityRoutines::MakeUPPERCase(state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentType(CompNum));
+                    auto foundType = static_cast<ValidEquipListType>(getEnumerationValue(validEquipNamesUC, typeNameUC));
```
Comment (amirroth) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:480]: Don't `auto` scalars. Also is `ValidEquipListType` a good name for an `enum class`? What about just `EquipType`? Aren't `Valid` and `List` implied here.

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 5]
```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
...
+
+                    case ValidEquipListType::FanSystemModel:
                         thisDOAS.FanName = CompName;
                         thisDOAS.m_FanTypeNum = SimAirServingZones::CompType::Fan_System_Object;
                         thisDOAS.m_FanIndex = HVACFan::getFanObjectVectorIndex(state, CompName);
-                        state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).InletNodeNum(CompNum) =
-                            state.dataHVACFan->fanObjs[thisDOAS.m_FanIndex]->inletNodeNum;
-                        if (state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).InletNodeNum(CompNum) == 0) {
+                        thisInletNodeNum = state.dataHVACFan->fanObjs[thisDOAS.m_FanIndex]->inletNodeNum;
+                        if (thisInletNodeNum == 0) {
```
Comment (amirroth) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:513]: Can use `InletNodeErrFlag |= (thisInletNodeNum == 0);`

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 6]
```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
...
-                                                         InletNodeErrFlag);
-                        state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).OutletNodeNum(CompNum) =
-                            WaterCoils::GetCoilOutletNode(state,
-                                                          SELECT_CASE_var,
-                                                          state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum),
-                                                          OutletNodeErrFlag);
+                        break;
+
+                    case ValidEquipListType::CoilCoolingWater:
+                        thisInletNodeNum = WaterCoils::GetCoilInletNode(state, typeNameUC, CompName, InletNodeErrFlag);
```
Comment (amirroth) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:557]: Here is how I would rank the implementation options for what is going on here.

1. Calling a single function that searches for the object and returns either its index or a reference to it and then grabbing the fields you need directly.
2. Calling a single function that searches for the object and returns a `std::pair`.
3. Calling two functions that search for the same object and but pull different fields off of it.

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 7]
```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
...
-                                                         InletNodeErrFlag);
-                        state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).OutletNodeNum(CompNum) =
-                            WaterCoils::GetCoilOutletNode(state,
-                                                          SELECT_CASE_var,
-                                                          state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum),
-                                                          OutletNodeErrFlag);
+                        break;
+
+                    case ValidEquipListType::CoilHeatingWater:
+                        thisInletNodeNum = WaterCoils::GetCoilInletNode(state, typeNameUC, CompName, InletNodeErrFlag);
```
Comment (amirroth) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:571]: Let me also say something about these passed error flags. There is no need for a separate undifferentiated error flag (i.e., the flag doesn't say what the error _was_ only that there was one) if there are one or more return values that can be interpreted to mean error. In C (and subsequently C++) the value `-1` or the pointer value `nullptr` are generally returned when something is not found. `thisInletNodeNum = -1` has the same information in it as `thisInletNodeNum = -1` and `InletNodeErrFlag = true`. Just something to think about.

File: src/EnergyPlus/AirLoopHVACDOAS.cc (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:471]: Got it.  In this case, CompType and CompName are both strings, and they are used in function calls that don't yet accept string_view, so I think the best path forward right now is to just use string references which at least avoid the copy.  For int, enum, bool, etc., yep, I will definitely try not to use references.

File: src/EnergyPlus/AirLoopHVACDOAS.cc (hunk 2 above)
Comment (Myoldmopar) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:472]: :+1:

File: src/EnergyPlus/AirLoopHVACDOAS.cc (hunk 4 above)
Comment (Myoldmopar) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:480]: The object being searched here is valled *:EquipmentList, so the List part is not indicating the list of enum values, it's the component name.  The Valid part is indicating that these are the component types that are allowed on this component input field.  I could take that off, but I thought it was a reasonable shorthand for "Valid component types allowed on this EquipmentList object".

File: src/EnergyPlus/AirLoopHVACDOAS.cc (hunk 5 above)
Comment (Myoldmopar) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:513]: Happy to do this, although, with the current code, I don't have to think about it as I read it...if it's zero, it sets the error flag.  With this operator, I have to think about what happens to it if it was already true or false, and I'm not sure it helps the readability.  

Certainly in cases where the value of the variable is explicitly set to true or false based on the condition, this is the way to go: `errFlag = (nodeNum == 0)`.  But adding in the OR-EQ makes it a little harder to grok.

I could be the only one with this opinion, and I am happy to make this change if you feel it is worth doing that.

File: src/EnergyPlus/AirLoopHVACDOAS.cc (hunk 6 above)
Comment (Myoldmopar) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:557]: What about if all these component objects derived from a single AirLoopComponent class, implemented a virtual method called `std::pair<int, int> getAirLoopNodes()`, and we kept them in a single array of components, or maybe better if we just kept pointers to the instances like we do in plant, then we could easily just loop over the components and call the virtual method on each one.

**Review by Myoldmopar — COMMENTED** [review]
Alright, I think I addressed all the commits on the evap cooler commit log as well as on the PR files changed tab.  This one should be ready to go once I finish testing and CI is happy.  Thanks for all the review comments @amirroth, hopefully I'm just doing most of the patterns right away from now on.

File: src/EnergyPlus/AirLoopHVACDOAS.cc (hunk 6 above)
Comment (amirroth) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:557]: That's some next level C++ 💩 !

File: src/EnergyPlus/AirLoopHVACDOAS.cc (hunk 5 above)
Comment (amirroth) [reply src/EnergyPlus/AirLoopHVACDOAS.cc:513]: So ... 

- `Y = X` is the same as `if(X) Y = true; else Y = false;`
- `Y ||= X` is the same as `if(X) Y = true;` should use `||=` for logical and `|= for bitwise. 
- `Y &&= X` is the same as `else Y = false;`

If and when you get brave enough to use them, you can add a comment that says "this is shorthand for whatever".

**Comment by Myoldmopar** [comment]
It's a bit hard to track all the comments here.  Honestly, my next progression on GitHub should be doing better of marking conversations as resolved.  Anyway, I _think_ this is sufficiently ready to get merged in if the rest of Decent comes back clean later today.  I'm going to pivot to reviewing other PRs for a little while before coming back to this effort, as the PR list has grown quite a bit.

**Comment by Myoldmopar** [comment]
I just built develop and this branch locally and ran regressions.  That file is not throwing BND diffs, I think it was because the branch was slightly behind develop.  I plan to merge this shortly unless anyone speaks up otherwise.

**Comment by mitchute** [comment]
I built and ran unit tests in release and debug here and it all ran fine. I'll lean on Edwin's regressions runs, as reported above. This is ready. Merging.

**Pull request merged**